src/codeu/chat/server/LocalFile.java

```diff
@@ -0,0 +1,170 @@
+package codeu.chat.server;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.ObjectOutputStream;
+import java.io.RandomAccessFile;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.FileInputStream;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Logger;
+import codeu.chat.util.Uuid;
+
+public class LocalFile implements Serializable
+{
+    private static final long serialVersionUID = 1L;
+    //Instance varibles for saving the current data of server.
+    private final HashMap<Uuid,User> users;
+    private final HashMap<Uuid,ConversationHeader> conversationHeaders;
+    private final HashMap<Uuid,Message> messages;
+
+    private transient final File file;
+
+    private static final Logger.Log LOG = Logger.newLog(LocalFile.class);
+
+    private transient boolean hasModified = false;//It indicates if there is a new data should be handled.
+    public LocalFile(File file)
+    {
+        this.file = file;
+        LocalFile newClass;
+        HashMap<Uuid,User>tempUsers = new HashMap<>();
+        HashMap<Uuid,ConversationHeader>tempConversationHeaders= new HashMap<>();
+        HashMap<Uuid,Message>tempMessages= new HashMap<>();
+        //Deserialization
+        try(FileInputStream fileInputStream = new FileInputStream(file);
+		    ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream))
+        {
+            //Read the information to current instance
+		    newClass = (LocalFile)objectInputStream.readObject();
```

Comment:
This is strange for a class to create new instances of itself by reading them in from an ObjectInputStream like this. Actually, in general it's not recommended to use java object serialization like this at all. Here's an article on this with more on why: https://softwareengineering.stackexchange.com/questions/191269/java-serialization-advantages-and-disadvantages-use-or-avoid

Instead, I'd recommend using the custom serializers that are already available in the (Message, Conversation, and User classes)[https://github.com/vgomezASC/CodeU-Summer-2017/blob/master/src/codeu/chat/common/Message.java#L28]. These serializers provide the same functionality, and you can use them to read/write these objects to and from a FileInputStream and FileOutputStream.